compiler/modules/nback/scaffold.bal

```diff
@@ -37,20 +37,37 @@ const BASE_REPR_TAGGED = "BASE_REPR_TAGGED";
 type BaseRepr UniformBaseRepr|BASE_REPR_TAGGED;
 type RetBaseRepr BaseRepr|BASE_REPR_VOID;
 
-type UniformRepr readonly & record {|
-    UniformBaseRepr base;
-    llvm:SingleValueType llvm;
-    t:UniformTypeBitSet subtype?;
+// Maps int to i1
+type BooleanRepr readonly & record {|
+    BASE_REPR_BOOLEAN base = BASE_REPR_BOOLEAN;
+    LLVM_BOOLEAN llvm = LLVM_BOOLEAN;
+    true isImmediate = true;
+|};
+
+// Maps int to i64
+type IntRepr readonly & record {|
+    BASE_REPR_INT base = BASE_REPR_INT;
+    LLVM_INT llvm = LLVM_INT;
+    t:IntSubtypeConstraints? constraints;
+    true isImmediate = true;
+|};
+
+// Maps float to llvm double
+type FloatRepr readonly & record {|
+    BASE_REPR_FLOAT base = BASE_REPR_FLOAT;
+    LLVM_DOUBLE llvm = LLVM_DOUBLE;
+    false isImmediate = false;
 |};
 
 // Maps any Ballerina value to a tagged pointer
 type TaggedRepr readonly & record {|
-    BASE_REPR_TAGGED base;
+    BASE_REPR_TAGGED base = BASE_REPR_TAGGED;
     t:UniformTypeBitSet subtype;
-    llvm:IntegralType llvm;
+    llvm:IntegralType llvm = LLVM_TAGGED_PTR;
+    boolean isImmediate;
```

Comment:
I think `alwaysImmediate` is better: since when this is false it means there can be both immediate and non-immediate values.